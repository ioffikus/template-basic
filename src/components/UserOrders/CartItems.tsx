import React from 'react';
import { Link } from 'react-router-dom';
import { Table } from 'antd';
import _ from 'lodash';
import Price from 'src/components/Price';
import CartProductImage from 'src/components/CartProductImage';

interface IRecord {
  key: number;
  pictureUrl: string;
  name: string;
  count: number;
  price: number;
  currency: string;
  uid: string;
}

interface IProps {
  ui: Alicanto.Models.UI;
  items: Alicanto.Models.CatalogItem[];
}

interface IState {}

class CartItems extends React.Component<IProps, IState> {
  getDataSource = (): IRecord[] => {
    return this.props.items.map(item => {
      const key = item.id;
      const uid = item.uid;
      const schemaFields = item.schema_fields;
      const pictureUrl = _.get(schemaFields, 'picture_url.value', '');
      const name = _.get(schemaFields, 'name.value', '');
      const count = _.get(item, 'count', 0);
      const currency = _.get(schemaFields, 'currency.value', '');
      const price = _.get(schemaFields, 'price.value', 0);
      return { key, pictureUrl, name, count, price, currency, uid };
    });
  };

  render() {
    const commonUI = this.props.ui.common;
    const columns = [
      {
        title: commonUI.tables.cells.image.i18n.defaultText,
        key: 'image',
        render: (text: string, record: IRecord) => {
          return <CartProductImage pictureUrl={record.pictureUrl} imageAlt={record.name} />;
        },
      },
      {
        title: commonUI.tables.cells.name.i18n.defaultText,
        key: 'name',
        render: (text: string, record: IRecord) => {
          return <Link to={`/product/${record.uid}`}>{record.name}</Link>;
        },
      },
      { title: commonUI.tables.cells.count.i18n.defaultText, dataIndex: 'count' },
      {
        title: commonUI.tables.cells.price.i18n.defaultText,
        dataIndex: 'price',
        render: (text: string, record: IRecord) => {
          const { price, currency } = record;
          return <Price price={price} currency={currency} />;
        },
      },
    ];

    return (
      <div className="root">
        <style jsx>{`
          .root :global(.ant-table-row:last-child td) {
            border: 0;
          }
        `}</style>
        <Table columns={columns} dataSource={this.getDataSource()} pagination={false} />
      </div>
    );
  }
}

export default CartItems;
