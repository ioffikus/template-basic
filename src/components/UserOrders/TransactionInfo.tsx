import React from 'react';

interface IProps {
  ui: Alicanto.Models.UI;
  details: string;
  status: string;
}

interface IState {}

class TransactionInfo extends React.Component<IProps, IState> {
  render() {
    const transactionUI = this.props.ui.routes.user.orders.transaction;
    const { details, status } = this.props;
    return (
      <div className="root">
        <style jsx>{`
          .root {
            font-size: 10px;
            font-weight: 500;
          }
          .name {
            margin-right: 10px;
            color: #b2bdcb;
            font-size: 9px;
          }
        `}</style>
        {status && (
          <div>
            <span className="name">{transactionUI.status.i18n.defaultText} : </span>
            <span>{status}</span>
          </div>
        )}
        {details && (
          <div>
            <span className="name">{transactionUI.details.i18n.defaultText} : </span>
            <span>{details}</span>
          </div>
        )}
      </div>
    );
  }
}

export default TransactionInfo;
