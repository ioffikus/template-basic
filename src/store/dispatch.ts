import { Store } from 'redux';
import _ from 'lodash';
import { showError } from 'src/store/common';

let store: Store<Store.IState>;

export const setupDispatch = (config: { store: Store<Store.IState> }) => {
  store = config.store;
};

const dispatch = (action: any) => {
  if (_.isFunction(action)) {
    return store.dispatch(action).catch((error: any) => {
      if (!_.isNull(error)) {
        showError(error);
      }
      return Promise.reject(error);
    });
  }
  return store.dispatch(action);
};

export default dispatch;
