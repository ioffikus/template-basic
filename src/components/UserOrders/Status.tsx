import React from 'react';
import { Popover } from 'antd';
import TransactionInfo from './TransactionInfo';

interface IProps {
  status: string;
  transactionDetails: string;
  transactionStatus: string;
  ui: Alicanto.Models.UI;
  getRootRef: () => HTMLElement;
}

interface IState {}

class Status extends React.Component<IProps, IState> {
  render() {
    const { status, transactionDetails, transactionStatus, getRootRef, ui } = this.props;
    const isTransaction = transactionDetails || transactionStatus;
    return (
      <div className="root">
        <style jsx>{`
          .root :global(.icon) {
            width: 17px;
            height: 17px;
            text-align: center;
            margin: 0 10px;
            display: inline-block;
            font-size: 0.8em;
            font-weight: bold;
            background-color: #e5ecf4;
            border-radius: 50%;
            cursor: pointer;
          }
        `}</style>
        {`Status: ${status}`}
        {isTransaction && (
          <Popover
            overlayClassName="status-popover"
            getPopupContainer={getRootRef}
            content={<TransactionInfo ui={ui} details={transactionDetails} status={transactionStatus} />}
          >
            <span className="icon">i</span>
          </Popover>
        )}
      </div>
    );
  }
}

export default Status;
