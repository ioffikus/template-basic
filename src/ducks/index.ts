// instead this, we have router in store via connected-react-router
// import { routerReducer as routing, RouterState } from 'react-router-redux';
import { combineReducers } from 'redux';
import auth from './auth';
import cart from './cart';
import cartPaidItems from './cartPaidItems';
import catalogQuery from './catalogQuery';
import catalog from './catalog';
import categories from './categories';
import categoriesMenu from './categoriesMenu';
import filter from './filter';
import locale from './locale';
import meta from './meta';
import page from './page';
import product from './product';
import progress from './progress';
import popularItems from './popularItems';
import recommendItems from './recommendItems';
import routes from './routes';
import ui from './ui';
import user from './user';
import userDelivery from './userDelivery';
import userOrders from './userOrders';

export default combineReducers<Store.IState>({
  ui,
  auth,
  user,
  categories,
  categoriesMenu,
  catalogQuery,
  catalog,
  product,
  progress,
  routes,
  meta,
  filter,
  cart,
  cartPaidItems,
  userDelivery,
  userOrders,
  popularItems,
  recommendItems,
  page,
  locale,
});
