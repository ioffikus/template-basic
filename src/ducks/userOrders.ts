import { createAction, createReducer } from 'redux-act';
import _ from 'lodash';
import { BusinessLogicError } from 'src/store/common';
import { deleteCart } from 'src/ducks/cart';
import { AxiosResponse } from 'axios';
import { push } from 'connected-react-router';
import { GET, POST } from 'src/core/API';
import EndpointsConfig from 'src/configs/endpoints';

export const REDUCER = 'USER_ORDERS';
const NS = `${REDUCER}__`;

export interface IUserOrdersState extends Alicanto.API.GET_USER_ORDERS_RESPONSE, Store.IReduxCommonState {
  currentPaymentId: number;
}

export const initialState: IUserOrdersState = {
  count: 0,
  next: null,
  previous: null,
  results: [],
  currentPaymentId: null,
  isRequestError: false,
  isRequesting: false,
};

const reducer = createReducer<IUserOrdersState>({}, initialState);

const getFailure = createAction(`${NS}GET_FAILURE`);
reducer.on(getFailure, state => ({
  ...state,
  isRequestError: true,
  isRequesting: false,
}));

const getSuccess = createAction<Alicanto.API.GET_USER_ORDERS_RESPONSE>(`${NS}GET_SUCCESS`);
reducer.on(getSuccess, (state, { next, previous, count, results }) => ({
  ...state,
  next,
  previous,
  count,
  results: [...results],
  isRequestError: false,
  isRequesting: false,
}));

const getRequest = createAction(`${NS}GET_REQUEST`);
reducer.on(getRequest, state => ({
  ...state,
  isRequestError: false,
  isRequesting: true,
}));

const postFailure = createAction(`${NS}POST_FAILURE`);
reducer.on(postFailure, state => ({
  ...state,
  count: 0,
  next: null,
  previous: null,
  results: [],
  currentPaymentId: null,
  isRequestError: true,
  isRequesting: false,
}));

const postSuccess = createAction(`${NS}POST_SUCCESS`);
reducer.on(postSuccess, state => ({
  ...state,
  isRequestError: false,
  isRequesting: false,
}));

const postRequest = createAction(`${NS}POST_REQUEST`);
reducer.on(postRequest, state => ({
  ...state,
  isRequestError: false,
  isRequesting: true,
}));

const setCurrentPaymentId = createAction<number>(`${NS}SET_CURRENT_PAYMENT_ID`);
reducer.on(setCurrentPaymentId, (state, currentPaymentId) => ({
  ...state,
  currentPaymentId,
}));

interface IUserOrdersQuery {
  page?: string;
  page_size?: string;
}

const getUserOrdersRequestParams = ({ page, page_size }: IUserOrdersQuery) => {
  const params: IUserOrdersQuery = {
    page_size: '3',
  };
  if (!_.isNil(page)) {
    params.page = page;
  }
  if (!_.isNil(page_size)) {
    params.page_size = page_size;
  }
  return params;
};

export const getUserOrders = (page = '1'): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    dispatch(getRequest());
    GET(
      {
        url: `${EndpointsConfig.USER_ORDERS}`,
        params: getUserOrdersRequestParams({ page }),
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        const data = resp.data as Alicanto.API.GET_USER_ORDERS_RESPONSE;
        let results = data.results;
        results = results.map((item: Alicanto.Models.Order) => ({
          ...item,
          cart: {
            ...item.cart,
            total: parseFloat(_.get(item.cart, 'total', '0')),
          },
        }));
        if (page !== '1') {
          const oldResults = getState().userOrders.results;
          results = [...oldResults, ...results];
        }
        dispatch(getSuccess({ ...data, results }));
        resolve();
      })
      .catch(error => {
        dispatch(getFailure());
        reject(error);
      });
  }).then(() => {
    const currentPaymentId = getState().userOrders.currentPaymentId;
    if (process.env.BROWSER && _.isNull(currentPaymentId)) {
      const lastPaymentTimeInMs = window.localStorage.getItem('paymentTime');
      const lastPaymentId = window.localStorage.getItem('paymentId');
      if (lastPaymentTimeInMs && lastPaymentId) {
        const msNow = Date.now();
        const passingTime = msNow - parseInt(lastPaymentTimeInMs, 10);
        if (passingTime < 30 * 1000) {
          dispatch(setCurrentPaymentId(parseInt(lastPaymentId, 10)));
          setTimeout(() => {
            dispatch(setCurrentPaymentId(null));
            window.localStorage.removeItem('paymentTime');
            window.localStorage.removeItem('paymentId');
          }, 30 * 1000 - passingTime);
        }
      }
    }
  });
};

export const checkoutUserOrder = (): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    const isAuthenticated = getState().auth.isAuthenticated;
    if (!isAuthenticated) {
      const anonCheckout = getState().ui.routes.cart.errors.anonCheckout.i18n.defaultText;
      const error = new BusinessLogicError(anonCheckout);
      reject(error);
      return;
    }
    dispatch(postRequest());
    const data: Alicanto.API.POST_USER_ORDERS_BODY = { cart: getState().cart.id };
    POST(
      {
        data,
        url: EndpointsConfig.USER_ORDERS,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        dispatch(deleteCart());
        dispatch(postSuccess());
        if (process.env.BROWSER) {
          //  TODO: move to external?
          dispatch(push('/user/orders'));
        }
        resolve();
      })
      .catch(error => {
        dispatch(postFailure());
        reject(error);
      });
  });
};

export const payAnOrder = (id: number): Store.IReduxThunkCb => (dispatch, getState, xShopHeaders) => {
  return new Promise((resolve, reject) => {
    const currentPaymentId = getState().userOrders.currentPaymentId;
    if (currentPaymentId) {
      const ui = getState().ui;
      reject(new BusinessLogicError(ui.routes.user.orders.paymentWarning.i18n.defaultText));
      return;
    }
    if (process.env.BROWSER) {
      window.localStorage.setItem('paymentTime', '' + Date.now());
      window.localStorage.setItem('paymentId', `${id}`);
    }
    dispatch(setCurrentPaymentId(id));
    setTimeout(() => {
      dispatch(setCurrentPaymentId(null));
      if (process.env.BROWSER) {
        window.localStorage.removeItem('paymentTime');
        window.localStorage.removeItem('paymentId');
      }
    }, 30 * 1000);
    dispatch(postRequest());
    const data: Alicanto.API.POST_PAYMENTS_PAY_BODY = { order: id + '' };
    POST(
      {
        data,
        url: EndpointsConfig.PAY_ORDER,
      },
      xShopHeaders,
    )
      .then((resp: AxiosResponse) => {
        dispatch(postSuccess());
        const data = resp.data as Alicanto.API.POST_PAYMENTS_PAY_RESPONSE;
        const redirect_url = data.redirect_to.redirect_url;
        if (process.env.BROWSER) {
          window.location.assign(redirect_url);
        }
        resolve();
      })
      .catch(error => {
        dispatch(postFailure());
        reject(error);
      });
  });
};

export default reducer;
