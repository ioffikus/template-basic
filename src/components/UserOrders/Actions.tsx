import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'antd';
import dispatch from 'src/store/dispatch';
import { payAnOrder } from 'src/ducks/userOrders';

interface IOwnProps {
  isPaid: boolean;
  id: number;
}

interface IConnectedState {
  ui: Alicanto.Models.UI;
  currentPaymentId: number;
}

type IProps = IOwnProps & IConnectedState;

interface IState {}

const mapStateToProps = (state: Store.IState) => ({
  ui: state.ui,
  currentPaymentId: state.userOrders.currentPaymentId,
});

const WrappedActions = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Actions extends React.Component<IProps, IState> {
    handleClick = () => {
      const { id } = this.props;
      dispatch(payAnOrder(id));
    };

    render() {
      const { isPaid, id, currentPaymentId, ui } = this.props;
      const buttonUI = ui.routes.user.orders.button.i18n.defaultText;
      return (
        <div className="root">
          {!isPaid && (
            <Button
              type="primary"
              size="small"
              onClick={this.handleClick}
              key="button"
              loading={currentPaymentId === id}
            >
              {buttonUI}
            </Button>
          )}
        </div>
      );
    }
  },
);

export default WrappedActions;
