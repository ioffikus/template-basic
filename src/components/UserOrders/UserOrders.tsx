import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import { Table, Spin, Button } from 'antd';
import moment from 'moment';
import dispatch from 'src/store/dispatch';
import { getUserOrders } from 'src/ducks/userOrders';
import CartItems from './CartItems';
import Status from './Status';
import Actions from './Actions';
import Price from 'src/components/Price';

interface IRecord {
  key: number;
  id: number;
  status: string;
  date: string;
  itemsCount: number;
  total: number;
  currency: string;
  isPaid: boolean;
  cart: Alicanto.Models.Cart;
  transactionDetails: string;
  transactionStatus: string;
}

interface IOwnProps {}

interface IConnectedState {
  ui: Alicanto.Models.UI;
  locale: string;
  items: Alicanto.Models.Order[];
  isRequesting: boolean;
  hasNextPage: boolean;
}

type IProps = IOwnProps & IConnectedState;

interface IState {
  isMounted: boolean;
  page: number;
}

const mapStateToProps = (state: Store.IState) => ({
  ui: state.ui,
  locale: state.locale.current,
  items: state.userOrders.results,
  isRequesting: state.userOrders.isRequesting,
  hasNextPage: !!state.userOrders.next,
});

const WrappedOrders = connect<IConnectedState, undefined, IOwnProps>(mapStateToProps)(
  class Orders extends React.Component<IProps, IState> {
    rootRef: HTMLElement;

    state: IState = {
      isMounted: false,
      page: 1,
    };

    setRootRef = (element: HTMLElement) => (this.rootRef = element);

    getRootRef = () => {
      return this.rootRef;
    };

    getDataSource = (): IRecord[] => {
      return this.props.items.map(item => {
        const key = item.id;
        const id = item.id;
        const cart = item.cart;
        const date = moment(cart.modified)
          .locale(this.props.locale)
          .format('lll');
        const itemsCount = cart.items_count;
        const total = cart.total;
        // TODO ch1333
        const currency = _.get(item, 'cart.items[0].schema_fields.currency.value', '');
        const status = item.status;
        const isPaid = item.is_paid;
        const transactionDetails = item.transaction_details;
        const transactionStatus = item.transaction_status;

        return {
          key,
          id,
          itemsCount,
          total,
          currency,
          status,
          date,
          isPaid,
          cart,
          transactionDetails,
          transactionStatus,
        };
      });
    };

    renderCartItems = (record: IRecord) => {
      const items = record.cart.items;
      return <CartItems ui={this.props.ui} items={items} />;
    };

    handleShowMore = () => {
      this.setState((prevState, props) => {
        const page = prevState.page + 1;
        setTimeout(() => dispatch(getUserOrders('' + page)));
        return {
          page,
        };
      });
    };

    componentDidMount() {
      dispatch(getUserOrders('' + this.state.page));
      this.setState({ isMounted: true });
    }

    render() {
      const userOrdersUI = this.props.ui.routes.user.orders;
      const commonUI = this.props.ui.common;
      const columns = [
        {
          title: commonUI.tables.cells.status.i18n.defaultText,
          key: 'status',
          render: (text: string, record: IRecord) => {
            const { status, transactionDetails, transactionStatus } = record;
            return (
              <Status
                ui={this.props.ui}
                status={status}
                transactionDetails={transactionDetails}
                transactionStatus={transactionStatus}
                getRootRef={this.getRootRef}
              />
            );
          },
        },
        { title: commonUI.tables.cells.date.i18n.defaultText, dataIndex: 'date' },
        { title: commonUI.tables.cells.itemsCount.i18n.defaultText, dataIndex: 'itemsCount' },
        {
          title: commonUI.tables.cells.total.i18n.defaultText,
          key: 'total',
          render: (text: string, record: IRecord) => {
            const { currency, total } = record;
            return <Price price={total} currency={currency} />;
          },
        },
        {
          title: commonUI.tables.cells.actions.i18n.defaultText,
          key: 'actions',
          render: (text: string, record: IRecord) => {
            const { isPaid, id } = record;
            return <Actions isPaid={isPaid} id={id} />;
          },
        },
      ];
      const defaultExpandedRowKeys = this.props.items.length ? [this.props.items[0].id] : [];
      return (
        <div className="root" ref={this.setRootRef}>
          <style jsx>{`
            .root {
              font-size: 14px;
              font-weight: bold;
              padding-bottom: 15px;
            }
            .root :global(.ant-table-thead > tr > th) {
              color: #062f49;
              font-weight: 500;
              text-transform: uppercase;
              background: transparent;
            }
            .root :global(.main-table) {
              background: #fff;
              margin-bottom: 16px;
            }
            .footer {
              text-align: center;
            }
            .root :global(.status-popover) {
              max-width: 400px;
            }
            .root :global(.status-popover .ant-popover-arrow) {
              display: block;
            }
          `}</style>
          {this.state.isMounted &&
            (_.isEmpty(this.props.items) ? (
              !this.props.isRequesting && (
                <div className="app-text-empty text-empty">{userOrdersUI.emptyStatus.i18n.defaultText}</div>
              )
            ) : (
              <Table
                className="main-table"
                columns={columns}
                dataSource={this.getDataSource()}
                pagination={false}
                expandedRowRender={this.renderCartItems}
                defaultExpandedRowKeys={defaultExpandedRowKeys}
              />
            ))}
          <div className="footer">
            {this.props.isRequesting ? (
              <Spin />
            ) : (
              this.props.hasNextPage && (
                <Button onClick={this.handleShowMore} size="large" className="app-show-more-button">
                  {commonUI.buttons.showMore.i18n.defaultText}
                </Button>
              )
            )}
          </div>
        </div>
      );
    }
  },
);

export default WrappedOrders;
