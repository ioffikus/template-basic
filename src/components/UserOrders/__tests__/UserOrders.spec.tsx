import React from 'react';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import moment from 'moment-timezone';
import { Spin, Button } from 'antd';
import { setupDispatch } from 'src/store/dispatch';
import uiStateMock from 'src/store/__mocks__/uiStore';
import localeStateMock from 'src/store/__mocks__/localeStore';
import userOrdersStateMock, { initialState as userOrdersInitialState } from 'src/store/__mocks__/userOrdersStore';
import { getUserOrders } from 'src/ducks/userOrders';
import UserOrders from '../UserOrders';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('src/ducks/userOrders', () => {
  return {
    getUserOrders: jest.fn(() => {
      return () => Promise.resolve();
    }),
  };
});

describe(null, () => {
  beforeEach(() => {
    moment.tz.setDefault('Europe/Moscow');
  });
  afterEach(() => {
    moment.tz.setDefault(null);
  });
  it('render() with props.items', () => {
    const store = mockStore({
      ui: uiStateMock,
      locale: localeStateMock,
      userOrders: userOrdersStateMock,
    });
    setupDispatch({ store });
    const wrapper = shallow(<UserOrders />, { context: { store } });
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('render() without props.items', () => {
    const store = mockStore({
      ui: uiStateMock,
      locale: localeStateMock,
      userOrders: userOrdersInitialState,
    });
    setupDispatch({ store });
    const wrapper = shallow(<UserOrders />, { context: { store } });
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('componentDidMount()', () => {
    const page = '1';
    const store = mockStore({
      ui: uiStateMock,
      locale: localeStateMock,
      userOrders: userOrdersStateMock,
    });
    setupDispatch({ store });
    expect(getUserOrders).toBeCalledWith(page);
  });
  it('Show Spin', () => {
    const store = mockStore({
      ui: uiStateMock,
      locale: localeStateMock,
      userOrders: { ...userOrdersStateMock, isRequesting: true },
    });
    setupDispatch({ store });
    const wrapper = shallow(<UserOrders />, { context: { store } });
    const dummyWrapper = wrapper.dive();
    const spinWrapper = dummyWrapper.find(Spin);
    const spinInspance = spinWrapper.dive().instance() as Spin;
    expect(spinWrapper).toHaveLength(1);
    expect(spinInspance.props.spinning).toBeTruthy();
  });
  it('Has "Show more" button ', () => {
    const store = mockStore({
      ui: uiStateMock,
      locale: localeStateMock,
      userOrders: { ...userOrdersStateMock, next: 'dummy' },
    });
    setupDispatch({ store });
    const wrapper = shallow(<UserOrders />, { context: { store } });
    const dummyWrapper = wrapper.dive();
    const buttonWrapper = dummyWrapper.find(Button);
    expect(buttonWrapper).toHaveLength(1);
  });
  it('Click on the "Show more" button ', () => {
    jest.useFakeTimers();
    const page = '2';
    const store = mockStore({
      ui: uiStateMock,
      locale: localeStateMock,
      userOrders: { ...userOrdersStateMock, next: 'dummy' },
    });
    setupDispatch({ store });
    const wrapper = shallow(<UserOrders />, { context: { store } });
    const dummyWrapper = wrapper.dive();
    const buttonWrapper = dummyWrapper.find(Button);
    const buttonInstance = buttonWrapper.dive().instance() as Button;
    buttonInstance.props.onClick(null);
    jest.runAllTimers();
    expect(getUserOrders).toBeCalledWith(page);
    jest.clearAllTimers();
    jest.useRealTimers();
  });
});
