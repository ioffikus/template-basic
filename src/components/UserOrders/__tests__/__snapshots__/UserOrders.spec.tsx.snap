// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() with props.items 1`] = `
<div
  className="jsx-1430125611 root"
>
  <JSXStyle
    css=".root.jsx-1430125611{font-size:14px;font-weight:bold;padding-bottom:15px;}.root.jsx-1430125611 .ant-table-thead > tr > th{color:#062f49;font-weight:500;text-transform:uppercase;background:transparent;}.root.jsx-1430125611 .main-table{background:#fff;margin-bottom:16px;}.footer.jsx-1430125611{text-align:center;}.root.jsx-1430125611 .status-popover{max-width:400px;}.root.jsx-1430125611 .status-popover .ant-popover-arrow{display:block;}"
    styleId="1430125611"
  />
  <Table
    bordered={false}
    className="main-table"
    columns={
      Array [
        Object {
          "key": "status",
          "render": [Function],
          "title": "status",
        },
        Object {
          "dataIndex": "date",
          "title": "date",
        },
        Object {
          "dataIndex": "itemsCount",
          "title": "items",
        },
        Object {
          "key": "total",
          "render": [Function],
          "title": "total",
        },
        Object {
          "key": "actions",
          "render": [Function],
          "title": "actions",
        },
      ]
    }
    dataSource={
      Array [
        Object {
          "cart": Object {
            "id": 33,
            "items": Array [
              Object {
                "count": 2,
                "id": 83,
                "in_stock": true,
                "is_deleted": false,
                "modified": "2017-12-15T11:23:43.370090Z",
                "schema_fields": Object {
                  "currency": Object {
                    "label": "Валюта",
                    "type": "string",
                    "value": "$",
                  },
                  "description": Object {
                    "label": "Описание",
                    "type": "text",
                    "value": "The New York Times bestselling series that inspired the international hit video game: The Witcher.

Geralt is a witcher, a man whose magic powers, enhanced by long training and a mysterious elixir, have made him a brilliant fighter and a merciless assassin. Yet he is no ordinary murderer: his targets are the multifarious monsters and vile fiends that ravage the land and attack the innocent. 

This is a collection of short stories, following the adventures of the hit collection THE LAST WISH. Join Geralt as he battles monsters, demons and prejudices alike...

Witcher collections
The Last Wish
Sword of Destiny

Witcher novels
Blood of Elves 
The Time of Contempt
Baptism of Fire
The Tower of Swallows
Lady of the Lake

The Malady and Other Stories: An Andrzej Sapkowski Sampler (e-only)


Translated from original Polish by David French",
                  },
                  "name": Object {
                    "label": "Имя",
                    "type": "string",
                    "value": "Sword of Destiny (The Witcher)",
                  },
                  "picture_url": Object {
                    "label": "Картинка",
                    "type": "string",
                    "value": "http://localhost:8000/media/item_pictures/2017-12-14T140837.2674370000dc74b408d40b4b7783ee3232c6554960.jpg",
                  },
                  "price": Object {
                    "label": "Цена",
                    "type": "decimal",
                    "value": 0.01,
                  },
                  "short_description": Object {
                    "label": "Краткое описание",
                    "type": "text",
                    "value": "The New York Times bestselling series that inspired the international hit video game: The Witcher.

",
                  },
                },
                "uid": "3cf12d0021c3438c9d43",
              },
            ],
            "items_count": 2,
            "modified": "2017-12-15T11:23:29.178841Z",
            "total": 0.02,
          },
          "currency": "$",
          "date": "Dec 15, 2017 2:23 PM",
          "id": 26,
          "isPaid": false,
          "itemsCount": 2,
          "key": 26,
          "status": "подтвержден",
          "total": 0.02,
          "transactionDetails": "",
          "transactionStatus": "",
        },
        Object {
          "cart": Object {
            "id": 32,
            "items": Array [
              Object {
                "count": 1,
                "id": 82,
                "in_stock": true,
                "is_deleted": false,
                "modified": "2017-12-15T11:23:21.985610Z",
                "schema_fields": Object {
                  "currency": Object {
                    "label": "Валюта",
                    "type": "string",
                    "value": "€",
                  },
                  "description": Object {
                    "label": "Описание",
                    "type": "text",
                    "value": "The New York Times bestselling series that inspired the international hit video game: The Witcher.

For over a century, humans, dwarves, gnomes, and elves have lived together in relative peace. But times have changed, the uneasy peace is over, and now the races are fighting once again. The only good elf, it seems, is a dead elf.

Geralt of Rivia, the cunning assassin known as The Witcher, has been waiting for the birth of a prophesied child. This child has the power to change the world - for good, or for evil.

As the threat of war hangs over the land and the child is hunted for her extraordinary powers, it will become Geralt's responsibility to protect them all - and the Witcher never accepts defeat.

Blood of Elves is the first full-length Witcher novel, and the perfect follow up if you've read The Last Wish collection.


Witcher novels
Blood of Elves 
The Time of Contempt
Baptism of Fire
The Tower of Swallows
Lady of the Lake

Witcher collections
The Last Wish
Sword of Destiny

The Malady and Other Stories: An Andrzej Sapkowski Sampler (e-only)


Translated from original Polish by Danusia Stok",
                  },
                  "name": Object {
                    "label": "Имя",
                    "type": "string",
                    "value": "Blood of Elves (The Witcher)",
                  },
                  "picture_url": Object {
                    "label": "Картинка",
                    "type": "string",
                    "value": "http://localhost:8000/media/item_pictures/2017-12-14T140838.10681000001414d5bc0446468394bf70a70a0ed503.jpg",
                  },
                  "price": Object {
                    "label": "Цена",
                    "type": "decimal",
                    "value": 0,
                  },
                  "short_description": Object {
                    "label": "Краткое описание",
                    "type": "text",
                    "value": "The New York Times bestselling series that inspired the international hit video game: The Witcher.

",
                  },
                },
                "uid": "5929a9e26a654ff785f5",
              },
            ],
            "items_count": 1,
            "modified": "2017-12-15T10:13:36.737706Z",
            "total": 0,
          },
          "currency": "€",
          "date": "Dec 15, 2017 1:13 PM",
          "id": 25,
          "isPaid": true,
          "itemsCount": 1,
          "key": 25,
          "status": "оплачен",
          "total": 0,
          "transactionDetails": "",
          "transactionStatus": "",
        },
      ]
    }
    defaultExpandedRowKeys={
      Array [
        26,
      ]
    }
    expandedRowRender={[Function]}
    indentSize={20}
    loading={false}
    locale={Object {}}
    pagination={false}
    prefixCls="ant-table"
    rowKey="key"
    rowSelection={null}
    showHeader={true}
    size="large"
    useFixedHeader={false}
  />
  <div
    className="jsx-1430125611 footer"
  />
</div>
`;

exports[`render() without props.items 1`] = `
<div
  className="jsx-1430125611 root"
>
  <JSXStyle
    css=".root.jsx-1430125611{font-size:14px;font-weight:bold;padding-bottom:15px;}.root.jsx-1430125611 .ant-table-thead > tr > th{color:#062f49;font-weight:500;text-transform:uppercase;background:transparent;}.root.jsx-1430125611 .main-table{background:#fff;margin-bottom:16px;}.footer.jsx-1430125611{text-align:center;}.root.jsx-1430125611 .status-popover{max-width:400px;}.root.jsx-1430125611 .status-popover .ant-popover-arrow{display:block;}"
    styleId="1430125611"
  />
  <div
    className="jsx-1430125611 app-text-empty text-empty"
  >
    No orders
  </div>
  <div
    className="jsx-1430125611 footer"
  />
</div>
`;
